src/main/kotlin/no/nav/tilleggsstonader/sak/vedtak/læremidler/beregning/LæremidlerBeregningService.kt

```diff
@@ -1,49 +1,132 @@
+
+        val aktiviteter = finnAktiviteter(behandlingId)
+
+        val beregningsresultatForMåned = vedtaksperioder.flatMap { vedtaksperiode ->
+            val utbetalingsperioder = vedtaksperiode.delTilUtbetalingsPerioder()
+
+            utbetalingsperioder.map { utbetalingsperiode ->
+                val relevantStønadsperiode = utbetalingsperiode.finnRelevantStønadsperiode(stønadsperioder)
+
+                lagBeregningsresultatForMåned(
+                    utbetalingsperiode = utbetalingsperiode,
+                    stønadsperiode = relevantStønadsperiode,
+                    aktivitet = utbetalingsperiode.finnRelevantAktivitet(
+                        aktiviteter = aktiviteter,
+                        aktivitetType = relevantStønadsperiode.aktivitet,
+                    ),
+                )
+            }
+        }
+
+        return BeregningsresultatLæremidler(beregningsresultatForMåned)
+    }
+
+    private fun lagBeregningsresultatForMåned(
+        utbetalingsperiode: UtbetalingsPeriode,
+        stønadsperiode: Stønadsperiode,
+        aktivitet: Aktivitet,
+    ): BeregningsresultatForMåned {
+        val grunnlagsdata =
+            lagBeregningsGrunnlag(
+                periode = utbetalingsperiode,
+                aktivitet = aktivitet,
+                målgruppe = stønadsperiode.målgruppe,
             )
-        }.map { it.second }
-        return BeregningsresultatLæremidler(
-            perioder = beregningsgrunnlagPerMåned,
+
+        return BeregningsresultatForMåned(
+            beløp = finnBeløpForStudieprosent(grunnlagsdata.sats, grunnlagsdata.studieprosent),
+            grunnlag = grunnlagsdata,
         )
     }
 
-    fun lagBeregningsGrunnlag(periode: BeregningPeriode, måned: YearMonth): Beregningsgrunnlag {
+    private fun lagBeregningsGrunnlag(
+        periode: UtbetalingsPeriode,
+        aktivitet: Aktivitet,
+        målgruppe: MålgruppeType,
+    ): Beregningsgrunnlag {
         return Beregningsgrunnlag(
-            måned = måned,
-            studienivå = periode.studienivå,
-            studieprosent = periode.studieprosent,
-            sats = finnSatsForStudienivå(måned, periode.studienivå),
+            fom = periode.fom,
+            tom = periode.tom,
+            studienivå = aktivitet.studienivå ?: throw IllegalStateException("Studienivå finnes ikke på aktiviteten"),
+            studieprosent = aktivitet.prosent,
+            sats = finnSatsForStudienivå(periode, aktivitet.studienivå),
+            utbetalingsMåned = periode.utbetalingsMåned,
+            målgruppe = målgruppe,
         )
     }
 
-    fun finnBeløpForStudieprosent(sats: Int, studieprosent: Int): Int {
+    private fun finnAktiviteter(behandlingId: BehandlingId): List<Aktivitet> {
+        return vilkårperiodeRepository.findByBehandlingIdAndResultat(behandlingId, ResultatVilkårperiode.OPPFYLT)
+            .tilAktiviteter()
+    }
+
+    private fun validerVedtaksperioder(
```

Comment:
Burde vi validere at man kun støtter en vedtaksperiode nå?
Uten å ha sjekket testene så virker det som at koden virker helt greit nå, hvis man ikke har flere vedtaksperioder.
Så kan neste iterasjon bli å legge til støtte for flere vedtaksperioder

Reply:
Sant, så lenge de ikke innenfor den samme måneden 😬 
Eks 1-3jan, 5-16 jan, 20-25jan